### DISCUSSION TITLE
Customizing: Is it possible to print the command no. in the prompt?

### DISCUSSION THREAD

**Discussion by schittli** [JanDeDobbeleer/oh-my-posh#3115] [Troubleshoot] [answered]
Good morning

I have a question to Customizing the prompt because I was not able to find any solution.
I guess oh-my-posh / go doesn't have access to the command history in the shell (I don't know go, but found no way how to do that).

**This is the use-case / problem to solve:**
In PowerShell it is very useful to display the command number in the prompt because one can easily re-call any previous command by using `#<Number>`. 
In the example below, `1`, `2`, `3`, `4` are the command numbers:
```PowerShell
C:\Temp
1 » $Files = Get-ChildItem -Recurse # Just do something to get a command history
C:\Temp
2 » "No. of Files: $($Files.Count)" # Just do something to get a command history
No. of Files: 96
C:\Temp
3 » "No. of commands in the history: $((Get-History).Count)"  # Now, the command history has 2 entries
No. of commands in the history: 2
C:\Temp
4 » 
```

Therefore, it would be very useful if we could display the value of the PowerShell command: `(Get-History).Count + 1` in the prompt.

Is it possible to do that?
I know that accessing the PowerShell command: `(Get-History).Count + 1` in oh-my-posh would break the shell portability, but I have no plan to use it in bash 😃 

Thanks a lot for any help!
Thomas

**Comment by JanDeDobbeleer** [comment, marked as answer]
@schittli yes, you can use `Set-PoshContext` to [populate an environment variable](https://ohmyposh.dev/docs/configuration/templates#environment-variables) to display this in any segment using the template:

```
function Set-EnvVar {
    $env:POSH_HISTORY_COUNT=$((Get-History).Count)
}
New-Alias -Name 'Set-PoshContext' -Value 'Set-EnvVar' -Scope Global -Force
```

And use as such:

```
"template": " {{ .Env.POSH_HISTORY_COUNT }} "
```

**Reply by schittli** [reply]
**Thank you very much**, @JanDeDobbeleer, for your support and for your detailed answer!

**Just a note for other readers:**
Of one would like to have the command number displayed in a segment, then one usually wants to display `(Get-History).Count + 1` because the current command is not yet in the history 😃 

Thanks a lot, kind regards,
Thomas
